core/src/main/java/com/linecorp/armeria/server/RequestTimeoutController.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server;
+
+import com.linecorp.armeria.common.Request;
+import com.linecorp.armeria.common.Response;
+
+/**
+ * A controller that sets a deadline or resets to a timeout when the current {@link Request}
+ * timeout setting is changed.
+ *
+ * <p>Note: This interface is meant for internal use by server-side protocol implementation to reschedule
+ * a timeout task when a user updates the request timeout configuration.
+ */
+public interface RequestTimeoutController {
+
+    /**
+     * Sets the amount of time, that is after the specified {@code deadlineNanos} from now, allowed
+     * until receiving the current {@link Request} and sending the corresponding {@link Response} completely.
+     *
+     * @param deadlineNanos the new deadline value in nanoseconds. {@code 0} if disabled.
+     */
+    void setDeadline(long deadlineNanos);
```

Comment:
Terminology is not right here. 'Deadline' means a point in time when a timeout occurs, i.e. timeout occurs when `System.nanoTime() >= deadlineNanos`. What you are doing here is: `realDeadlineNanos = startTimeNanos + userSpecifiedValue`.

Reply:
Ahh. I misunderstood the deadline meaning. Thanks for kind explanation. 🙏